README.md

```diff
@@ -66,14 +66,14 @@ Check out the [full, working example repo](http://github.com/ethanresnick/json-a
 
 # Core Concepts
 ## Resource Type Descriptions <a name="resource-type-descriptions"></a>
-The JSON-API spec is built around the idea of typed resource collections. For example, you can have a `"people"` collection and a `"companies"` collection. (By convention, type names are plural and lowercase.)
+The JSON-API spec is built around the idea of typed resource collections. For example, you can have a `"people"` collection and a `"companies"` collection. (By convention, type names are plural, lowercase, and dasherized.)
 
 To use this library, you describe the special behavior (if any) that resources of each type should have, and then register that description with a central `ResourceTypeRegistry`. Then the library takes care of the rest. A resource type description is simply an object with the following properties:
 
 - `urlTemplates`: an object containing url templates used to output the json for resources of the type being described. Currently, the supported keys are: `"self"`, which defines the template used for building [resource urls](http://jsonapi.org/format/#document-structure-resource-urls) of that type, and `"relationship"`, which defines the template that will be used for [relationship urls](http://jsonapi.org/format/#fetching-relationships).
 - `dbAdapter`: the [database adapter](#database-adapters) to use to find and update these resources. By specifying this for each resource type, different resource types can live in different kinds of databases.
 
-- <a name="before-render"></a>`beforeRender` (optional): a function called on each resource after it's found by the adapter but before it's sent to the client. This lets you do things like hide fields that some users aren't authorized to see.
+- <a name="before-render"></a>`beforeRender` (optional): a function called on each resource after it's found by the adapter but before it's sent to the client. This lets you do things like hide fields that some users aren't authorized to see. It receives `resource`, `req`, `res`, and `superFn` as input and either a transformed resource or a promise as output. Returning or resolving to `undefined` will remove the resource from the response.
```

Comment:
This is very good info to document, but it seems a bit out of place in the readme, which is about describing the library's overall capabilities more than documenting specific arguments.

Maybe you could create a `beforeRender` function in one of the resource descriptions in the example repo, and add a [jsdoc block]() on top of that `beforeRender` explaining all the arguments/return values. Then, this readme could just include a link to the line with that docbloc?